lib/tests/misc.nix

```diff
@@ -518,6 +518,30 @@ runTests {
     expected = false;
   };
 
+  testFindFirstExample1 = {
+    expr = findFirst (x: x > 3) 7 [ 1 6 4 ];
+    expected = 6;
+  };
+
+  testFindFirstExample2 = {
+    expr = findFirst (x: x > 9) 7 [ 1 6 4 ];
+    expected = 7;
+  };
+
```

Comment:
The code has a lot of potential for off by one errors now. Could you add some more happy cases, not just with the match in the middle, because that might be the easy case so to speak. Also extra cases for lengths 1 and 2 wouldn't hurt.